### DISCUSSION TITLE
Contribute, but no developer

### DISCUSSION THREAD

**Discussion by morris2009** [jmcollin78/solar_optimizer#41] [General]
Hi, so far I really like Solar Optimizer. I did have some challenges getting it started, and I would love to contribute. I am not a developer, s coding isn't something I would be able to do, but i could create some additional images/screenshots and maybe write some additional help for people that also want to try this.
I also encounter some questions. For instance, the different colors of the icon and the checkmark with the icon. I am not completely sure yet what each of those states mean. Giving me some short answers, and I would be glad to write that with some additional examples as a sort of manual.

**Comment by jmcollin78** [comment]
Hello @morris2009 ,

You are welcome ! Contributing is not only coding and I would be please to add your contributions.

Concerning the colors:
- The color of the icon is state: yellow, the device is on, gray it is not on
![Capture d’écran 2024-07-03 à 08 23 58](https://github.com/jmcollin78/solar_optimizer/assets/1712374/5b43819f-8eaf-4cd5-8f49-1d24503bd302) -> not active

![Capture d’écran 2024-07-03 à 08 25 18](https://github.com/jmcollin78/solar_optimizer/assets/1712374/390684e9-936a-4873-8eeb-1491c221b78d) -> active

- the badge says 2 things: if it is enabled by user and if it is ready (check_usable true).

Examples:
    - ![Capture d’écran 2024-07-03 à 08 25 58](https://github.com/jmcollin78/solar_optimizer/assets/1712374/999ff14f-a9ba-4e7e-bfb0-cb24872a2e06) -> not active, disabled by the user (Enable entit-y is not on),
    - ![Capture d’écran 2024-07-03 à 08 26 53](https://github.com/jmcollin78/solar_optimizer/assets/1712374/85c17b0b-f87a-441d-9957-cc4a128294eb) -> not active, enabled (check) but not ready (ie the check_usable is false)
    - ![Capture d’écran 2024-07-03 à 08 28 05](https://github.com/jmcollin78/solar_optimizer/assets/1712374/9cde2c92-7a49-4813-a03f-d831ed44cae1) -> not active, enabled and ready (check_usable is true)
    - ![Capture d’écran 2024-07-03 à 08 29 53](https://github.com/jmcollin78/solar_optimizer/assets/1712374/fa9c846f-1494-4dea-889c-30a7d6fbfdf1) -> active (icon), enabled (check mark), but waiting for the next period (see duration_min) to be usable by the algorithm.

I hope this helps.

**Reply by morris2009** [reply]
Yes it helps.
I had a test device created with only 10Watt and small wait times (like 1min) so I could already see a bit what happened. Together with your explanation i  think i get it.
I'll try to write a bigger piece of text around it in how I would explain it here in my house to the rest. I can share that to be added to the github, or a wikipart or whereever.
I do need to do some reading to see how i can share stuff on github :)
